frbpoppy/time_dists.py

```diff
@@ -31,12 +31,54 @@ def regular(rate=2, n_srcs=1, n_days=1, z=0):
     time = np.tile(time_range, (n_srcs, 1))
 
     # Add redshift
-    time *= (1+z)[:, np.newaxis]
+    time *= np.atleast_1d(1+z)[:, np.newaxis]
     time[(time > n_days)] = np.nan
 
     return time
 
 
+def cyclic(rate=2, n_days=1, n_srcs=1, period=1, frac=.1, z=0):
+    """Generate a series of uniform burst times within an activity cycle."
+
+    Args:
+        rate (float/array): Number of events per day
+        n_days (int): Number of days
+        n_srcs (int): Number of sources
+        period (float/array): Period of activity cycle (days)
+        frac (float/array): Fraction of activity cycle a source is active
+        z (float/array): Redshift of sources
+    """
+    # ensure arguments that may be floats or arrays are arrays (length is number of sources)
+    rate = np.atleast_1d(rate)
+    period = np.atleast_1d(period)
+    frac = np.atleast_1d(frac)
+    z = np.atleast_1d(z)
+
+    # get number of bursts to generate for each source
+    nburst_per_cycle = (rate * frac * period).astype(int)
+    nburst_per_source = ((n_days / period) * nburst_per_cycle).astype(int)
+
+    # generate burst arrival times within each active period
+    times = np.random.uniform(0, frac[:, np.newaxis] * period[:, np.newaxis],
+                              (n_srcs, nburst_per_source.max())).astype(np.float32)
```

Comment:
There's still a bug in here, as this generates unsorted bursts.

Reply:
Any chance you still would have time to look into this? :)

Reply:
I'll have a look; for most of the function it doesn't matter that the bursts aren't sorted, but it does for the redshift calculation and probably the output should be sorted anyway?